testsuite/cluster-test/src/experiments/performance_benchmark.rs

```diff
@@ -76,47 +71,51 @@ impl PerformanceBenchmarkParams {
 impl ExperimentParam for PerformanceBenchmarkParams {
     type E = PerformanceBenchmark;
     fn build(self, cluster: &Cluster) -> Self::E {
-        if self.is_fullnode {
-            let num_nodes = cluster.fullnode_instances().len();
-            let nodes_down = (num_nodes * self.percent_nodes_down) / 100;
-            let (down_instances, up_instances) = cluster.split_n_fullnodes_random(nodes_down);
-            Self::E {
-                down_instances: down_instances.into_fullnode_instances(),
-                up_instances: up_instances.into_fullnode_instances(),
-                percent_nodes_down: self.percent_nodes_down,
-                duration: Duration::from_secs(self.duration),
-                trace: self.trace,
-            }
-        } else {
-            let num_nodes = cluster.validator_instances().len();
-            let nodes_down = (num_nodes * self.percent_nodes_down) / 100;
-            let (down_instances, up_instances) = cluster.split_n_validators_random(nodes_down);
-            Self::E {
-                down_instances: down_instances.into_validator_instances(),
-                up_instances: up_instances.into_validator_instances(),
-                percent_nodes_down: self.percent_nodes_down,
-                duration: Duration::from_secs(self.duration),
-                trace: self.trace,
+        let all_fullnode_instances = cluster.fullnode_instances();
+        let num_nodes = cluster.validator_instances().len();
+        let nodes_down = (num_nodes * self.percent_nodes_down) / 100;
+        let (down, up) = cluster.split_n_validators_random(nodes_down);
+        let up_validators = up.into_validator_instances();
+        let fullnodes: Vec<_> = up_validators
+            .iter()
+            .map(|val| {
+                all_fullnode_instances
+                    .iter()
+                    .find(|x| val.peer_id().unwrap() == x.peer_id().unwrap())
```

Comment:
We generally discourage using `.unwrap` in code base (with few notable exceptions, like `.lock().unwrap()`).

It seems all callers just do `.unwrap()` on result, so we might as well just return `String` in `val.peer_id()` and panic there peer id is not found. In future we probably will just make peer_id a field on Instance (which is not possible now with --swarm mode, but I think we going to refactor that out)

Reply:
Done